### DISCUSSION TITLE
What is the recommended way to have automatic backups?

### DISCUSSION THREAD

**Discussion by nikolaydyankov** [pocketbase/pocketbase#2755] [Q&A] [answered]
Hi there,

Currently I'm using a very naive `aws s3 sync pb_data s3://...` every hour to backup my database to S3. From what I understand, this might result in data corruption if the database is used while the sync occurs.

I looked into the [backups API](https://pocketbase.io/docs/api-backups/), but I'm getting 404:

```javascript
import PocketBase from 'pocketbase'

const pb = new PocketBase('http://127.0.0.1:8090')
await pb.admins.authWithPassword('...', .'..')
await pb.backups.create(`backup.zip`)
```

The URL is correct, I'm using it in the actual web app. But even if I get it to work, I believe this action locks the database while it zips the data folder and any write operations would fail.

My only option seems to be snapshots of the server on which the web app is running. Am I missing something?

Thanks

**Comment by ganigeorgiev** [comment, marked as answer]
You can use the builtin automated backups functionality available with PocketBase v0.16+ - https://pocketbase.io/docs/going-to-production/#backup-and-restore (_note that the cron expression timezone in the UI is in UTC_).

>  But even if I get it to work, I believe this action locks the database while it zips the data folder and any write operations would fail.

Writes will not fail directly but they will queue up until the backup operation completes.

**Reply by nikolaydyankov** [reply]
That solves it, thanks. One small thing - `./pocketbase.exe update` returns unknown command. I'm running version 0.15.3, are there any breaking changes?

**Reply by ganigeorgiev** [reply]
The `update` command was added with [v0.16.0](https://github.com/pocketbase/pocketbase/releases/tag/v0.16.0). After upgrading to v0.16+ you should be able to use it as usual.


There shouldn't be any breaking changes between v0.15.3 and v0.16+ but make sure to have a backup of your `pb_data` just in case. For full changelog, you can refer to https://github.com/pocketbase/pocketbase/blob/master/CHANGELOG.md.

**Reply by nikolaydyankov** [reply]
I just saw the release notes from 0.16.4 release notes about the update on windows, I'll update it manually.

Do you keep a list of any breaking changes?

**Reply by ganigeorgiev** [reply]
> Do you keep a list of any breaking changes?

All release notes could be found in https://github.com/pocketbase/pocketbase/blob/master/CHANGELOG.md. Breaking changes are usually prefixed with "**!**".

**Reply by nikolaydyankov** [reply]
Thanks, that answers all my questions. So far working with PB has been a pleasure, thank you for developing it.
Also the auto backups to S3 is exactly what I was looking for, many thanks for implementing it.

**Comment by erikd234** [comment]
To future users, if you land here, litestream is another great option, it constantly streams mini changes of the db to a s3 bucket so theres always an update backup to restore if there are issues

**Reply by serdonke** [reply]
This should probably be added to the docs.
